File: benchmarks/refs/cpp/task_spawn.cpp
#include <chrono>
#include <cstdio>
#include <thread>

static const int ITERS = 50000;

int main() {
    auto t0 = std::chrono::steady_clock::now();
    for (int i = 0; i < ITERS; i++) {
        std::thread([] {}).join();
    }
    auto t1 = std::chrono::steady_clock::now();
    double secs = std::chrono::duration<double>(t1 - t0).count();
    double per_op_ns = (secs * 1e9) / ITERS;
    std::printf("BENCH|task_spawn_cpp|iters=%d|total_sec=%.6f|per_op_ns=%.2f\n", ITERS, secs, per_op_ns);
    return 0;
}

File: benchmarks/refs/zig/task_spawn.zig
const std = @import("std");

const ITERS: i64 = 50_000;

pub fn main() !void {
    var t0 = try std.time.Instant.now();
    var i: i64 = 0;
    while (i < ITERS) : (i += 1) {
        const t = try std.Thread.spawn(.{}, struct {
            fn run() void {}
        }.run);
        t.join();
    }
    const t1 = try std.time.Instant.now();
    const secs = @as(f64, @floatFromInt(t1.since(t0))) / 1e9;
    const per_op_ns = (secs * 1e9) / @as(f64, @floatFromInt(ITERS));
    std.debug.print("BENCH|task_spawn_zig|iters={}|total_sec={d:.6}|per_op_ns={d:.2}\n", .{ ITERS, secs, per_op_ns });
}
